packages/web/src/components/form-fields/HarmonyBaseTextField.tsx

```diff
@@ -0,0 +1,43 @@
+import { TextInput } from '@audius/harmony'
+import type { TextInputProps, PasswordInputProps } from '@audius/harmony'
+import { useField } from 'formik'
+import { ComponentType } from 'react'
+
+export type BaseHarmonyTextFieldProps<
+  T extends TextInputProps | PasswordInputProps
+> = T & {
+  name: string
+  /**
+   * Clears out field errors while the input is being changed for a small UX improvement
+   * @default true
+   */
+  clearErrorOnChange?: boolean
+  inputComponent?: ComponentType<T>
+}
+
+/** Building block for Harmony Text Field components. Not intended for general use. */
+export function HarmonyBaseTextField<
```

Comment:
this makes a lot of sense with how its written... thinking about whether there is another way to do it just cause this is quite intense....